mydemoproject/demoapp1/models/borrowed_book.py

```diff
@@ -0,0 +1,58 @@
+from django.db import models
+from ._helper.base_model import BaseModel
+from django.utils import timezone
+from .user import CustomUser
+from .book import Book
+from .inventory import Inventory
+
+
+class BorrowedBook(BaseModel):
+    """
+    If a user is deleted, their associated borrowed book entries with a null return date
+    must also be deleted. This is to prevent data inconsistency where a user has borrowed
+    a book but never returned it. We don't want such entries in our database.
+    """
+
+    borrow_id = models.AutoField(primary_key=True)
+    user = models.ForeignKey(CustomUser, on_delete=models.CASCADE,
+                             related_name='borrowed_books')
+    book = models.ForeignKey(Book, on_delete=models.CASCADE)
+    borrow_date = models.DateField(auto_now_add=True)
+    return_date = models.DateField(null=True)
```

Comment:
These should be datetime fields. The names can be `borrow_dttm` and `return_dttm`.